### PR TITLE
feat: add gatekeeper for main db, improve docs

### PR DESCRIPTION
**Pull request by pyphilia**
Important changes are in `gatekeeper.ts` and `postgres.ts` (`createGatekeeper` scope)
I added comments to highlight parts I've added.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,12 +69,45 @@ export AWS_ACCESS_KEY_ID="yourkey"
 export AWS_SECRET_ACCESS_KEY="yoursecret"
 
 # use the passwords that will be really used!
-export TF_VAR_GRAASP_DB_PASSWORD="password" 
+export TF_VAR_GRAASP_DB_PASSWORD="password"
 export TF_VAR_ETHERPAD_DB_PASSWORD="password"
 export TF_VAR_MEILISEARCH_MASTER_KEY="masterkey"
+export TF_VAR_GRAASP_DB_GATEKEEPER_KEY_NAME="gatekeeper-keyname"
```
Comment (pyphilia) [inline README.md:75]: I added this new variable.

File: README.md [hunk 2]
```diff
@@ -82,113 +116,122 @@ Once everything is created properly, you can let the CI handle the infrastructur
...
+  - File upload
+  - ...
+- EC2
+  - Install psql
+
+```shell
+  sudo yum update
+  sudo yum install postgresql15
+  psql --version
+```
```
Comment (pyphilia) [inline README.md:218]: Instructions.

File: constructs/ec2.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import { Construct } from 'constructs';
```
Comment (pyphilia) [inline constructs/ec2.ts:1]: New file.

File: constructs/postgres.ts [hunk 4]
```diff
@@ -1,105 +1,149 @@
...
+      // Do not use the `ingress` and `egress` directly on the SecurityGroup for limitations reasons
+      // See note on https://registry.terraform.io/providers/hashicorp/aws/5.16.1/docs/resources/security_group#protocol
+      new VpcSecurityGroupIngressRule(scope, `${name}-allow-gatekeeper`, {
+        referencedSecurityGroupId: gateKeeper.securityGroup.id, // allowed source security group
+        fromPort: dbPort,
+        ipProtocol: 'tcp',
+        securityGroupId: dbSecurityGroup.id,
+        toPort: dbPort,
       });
+    }
```
Comment (pyphilia) [inline constructs/postgres.ts:68]: Gatekeeper option.

File: main.ts [hunk 5]
```diff
@@ -95,216 +115,357 @@ class GraaspStack extends TerraformStack {
...
-      "graasp",
+      'graasp',
+      'graasp',
       dbPassword,
       vpc,
       backendSecurityGroup,
-      CONFIG[environment.env].enableGraaspDatabaseReplication
+      CONFIG[environment.env].enableGraaspDatabaseReplication,
+      undefined,
+      true
```
Comment (pyphilia) [inline main.ts:165]: Enable gatekeeper.

File: package.json [hunk 6]
```diff
@@ -20,10 +20,10 @@
...
   },
   "dependencies": {
-    "@cdktf/provider-aws": "17.0.4",
-    "cdktf": "^0.18.0",
-    "cdktf-cli": "^0.18.0",
-    "constructs": "^10.2.70"
+    "@cdktf/provider-aws": "18.0.5",
+    "cdktf": "^0.19.1",
+    "cdktf-cli": "^0.19.1",
+    "constructs": "^10.3.0"
```
Comment (pyphilia) [inline package.json:26]: Update deps.

File: .github/workflows/infrastructure-deploy-dev-on-merge.yml [hunk 7]
```diff
@@ -12,16 +12,14 @@ permissions:
...
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
-      - uses: actions/setup-node@v3
-        with:
-          node-version: 18
+      - uses: actions/setup-node@v4
```
Comment (pyphilia) [inline .github/workflows/infrastructure-deploy-dev-on-merge.yml:22]: Update versions.

**Review by spaenleh — APPROVED** [review]
LGTM!

**Pull request merged**